services/multilang-support/src/utils.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python3
+
+# Importing Tokenizer and Model library, and torch
+from transformers import AutoTokenizer, AutoModelForSeq2SeqLM
+import torch
+import time
+import logging
+
+# Configure the logging settings
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s [%(levelname)s]: %(message)s",
+    datefmt="%y-%m-%d %H:%M %Z",
+)
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.DEBUG)
+
+
+# A logging wrapper function/decorator
+def log(func):
+    """
+    Timing & logging decorator for functions.
+    """
+
+    def wrapper(*args, **kwargs):
+        start_time = time.time()
+        function_output = func(*args, **kwargs)
+        end_time = time.time()
+        elapsed_time = end_time - start_time
+        LOGGER.debug(f"\'{func.__name__}\' takes {elapsed_time:.3f} seconds")
```

Comment:
Considering the times shown in your demo, probably good to put this in ms.

Reply:
I just mean the weights for the model you're using. Usually there's a pretty clear line in the Dockerfile or Python code itself where these are downloaded ahead of time, and I didn't see one when I looked through.

Reply:
In this implementation, the model is loaded when the program/module is started (see `instantiate()`), so it's not downloaded ahead of time.
I will look into that, the weights concept is new to me at the moment 🤔

Reply:
Ah no worries. How long does it take to download with a clean container? This is the same way the TTS is set up, but we had to find a way to predownload it to avoid a huge delay when the first request is received.

Reply:
For fresh container, it takes around ~3s to download and get ready (excluding container build time). Sometimes it randomly takes up to ~10s to get ready, but it's very rare. 

I also noticed (and forgot to mention anywhere) that first request after initialising often takes more time to process compared to requests after that. A query similar to the one during demo takes ~1.5s to process on first req, but any similar queries after that only takes ~0.4s to 0.8s.